client/src/App.tsx

```diff
@@ -49,27 +21,16 @@ export function App() {
     };
   }, []);
 
-  if (error || !posts) {
-    return (
-      <div>
-        Error! {error instanceof Error ? error.message : 'Unknown error'}
-      </div>
-    );
-  }
-
   return (
     <>
       <Header isMobile={isMobile} />
-      <div className="flex h-full">
-        {!isMobile && <NavBar categories={categories} />}
-        <div className="flex mx-auto">
-          <div className="basis-full p-4">
-            {posts.map((post, index) => (
-              <Card key={post.title + index} post={post} />
-            ))}
-          </div>
-        </div>
-      </div>
+      <Routes>
+        <Route path="/" element={<Home isMobile={isMobile} />} />
+        <Route
+          path="/categories/:categoryName"
+          element={<Home isMobile={isMobile} />}
+        />
+      </Routes>
```

Comment:
This works for now, but I'm actually thinking if we want to refactor this in some other way to make this more seamless. The first thing that comes to my head, is rather than using a queryParam for the `category`, we can just use state inside of the `Home` component, and when you select one in the `NavBar` it'd call the event handler in `Home` and update state. Then we just use that to make the request.

That way we don't need to worry about having two different routes, but then we'd lose the functionality of being able to go to a specific category. It depends on how you want the flow to go. If you want a user to be able to go to `www.website.com/categories/Snacks` directly, we'd need to keep the route, if you are fine with not sending them to a specific category, then we can just keep it in state